src/LinuxTracing/LibunwindstackMultipleOfflineAndProcessMemory.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef LINUX_TRACING_LIBUNWINDSTACK_MULTIPLE_OFFLINE_AND_PROCESS_MEMORY_H_
+#define LINUX_TRACING_LIBUNWINDSTACK_MULTIPLE_OFFLINE_AND_PROCESS_MEMORY_H_
+
+#include <absl/base/casts.h>
+#include <unwindstack/Memory.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_linux_tracing {
+
+// This class is a "view" of a stack slice (some copy of process memory). It contains a bare pointer
+// to the array that actually owns the stack data.
+class StackSliceView {
+ public:
+  StackSliceView(uint64_t start_address, uint64_t size, const char* data)
+      : start_address_{start_address}, size_{size}, data_{data} {}
+
+  [[nodiscard]] uint64_t start_address() const { return start_address_; }
+  [[nodiscard]] uint64_t end_address() const { return start_address_ + size_; }
+  [[nodiscard]] uint64_t size() const { return size_; }
+  [[nodiscard]] const char* data() const { return data_; }
+
+ private:
+  uint64_t start_address_;
+  uint64_t size_;
+  const char* data_;
+};
+
+// This class is a thin layer around unwindstack::MemoryOfflineBuffer, that allows querying address
+// and size of the underlying offline memory as well as specifying the memory region as
+// `StackSliceView`.
+class LibunwindstackOfflineMemory : public unwindstack::Memory {
```

Comment:
So yeah, now this is entirely an implementation detail of `LibunwindstackMultipleOfflineAndProcessMemory`, so it should be private inside it.

Reply:
Done.